sdk/core/Azure.Core/src/HttpMessage.cs

```diff
@@ -80,15 +92,49 @@ public Response Response
         /// </summary>
         public TimeSpan? NetworkTimeout { get; set; }
 
-        internal void AddPolicies(RequestContext? context)
+        internal void ApplyRequestContext(RequestContext? context)
         {
-            if (context == null || context.Policies == null || context.Policies.Count == 0)
+            if (context == null)
             {
                 return;
             }
 
-            Policies ??= new(context.Policies.Count);
-            Policies.AddRange(context.Policies);
+            if (context.Policies?.Count > 0)
+            {
+                Policies ??= new(context.Policies.Count);
+                Policies.AddRange(context.Policies);
+            }
+
+            if (context.Classifier != null)
+            {
+                var classifier = ResponseClassifier as CompositeClassifier;
+                if (classifier != null)
+                {
+                    classifier.PerCallClassifier = context.Classifier;
```

Comment:
Should we validate that the property is not already populated with some classifier? i.e. that the apply method has not been called twice.